### PR TITLE
chore: make unused block hashes in testing header creation explicit

### PR DESCRIPTION
**Pull request by tungle-notional**
## Description

closes: #3960 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by tungle-notional** [comment]
@nghuyenthevinh2000

**Comment by tungle-notional** [comment]
@colin-axner

File: testing/values.go [hunk 1]
```diff
@@ -67,4 +67,6 @@ var (
 	MockRecvCanaryCapabilityName = mock.MockRecvCanaryCapabilityName
 
 	prefix = commitmenttypes.NewMerklePrefix([]byte("ibc"))
+
+	UnusedHash = []byte{0x00}
```
Comment (tropicaldog) [inline testing/values.go:71]: Should be `UnusedHash = tmhash.Sum([]byte{0x00})` instead.

**Comment by nghuyenthevinh2000** [comment]
Hi @crodriguezvega @colin-axner, 

I hope that the IBC team will support some of our engineer contributions to ibc-go by approving workflows. @tungle-notional is new to Cosmos, and he wants to put his engineering skill to test by learning to contribute to IBC - go. I will ensure that his contributions conform to conventional commits guidelines.

**Review by crodriguezvega — APPROVED** [review]
Thank you, @tungle-notional! Welcome to Cosmos and looking forward to more contributions from you!

**Pull request merged**